wandb/integration/yolov8/yolov8.py

```diff
@@ -0,0 +1,267 @@
+from typing import List, Optional
+
+from ultralytics.yolo.engine.model import YOLO
+from ultralytics.yolo.engine.trainer import BaseTrainer
+from ultralytics.yolo.utils import RANK
+from ultralytics.yolo.utils.torch_utils import get_flops, get_num_params
+from ultralytics.yolo.v8.classify.train import ClassificationTrainer
+
+import wandb
+from wandb.sdk.lib import telemetry
+
+
+class WandbCallback:
+    """
+    An internal YOLO model wrapper that tracks metrics, and logs models to Weights & Biases.
+    Usage:
+    ```python
+    from wandb.integration.yolov8.yolov8 import WandbCallback
+    model = YOLO("yolov8n.pt")
+    wandb_logger = WandbCallback(model,)
+    for event, callback_fn in wandb_logger.callbacks.items():
+        model.add_callback(event, callback_fn)
+    ```
+    """
+
+    def __init__(
+        self,
+        yolo: YOLO,
+        run_name: Optional[str] = None,
+        project: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        resume: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        A utility class to manage wandb run and various callbacks for the ultralytics YOLOv8 framework.
+        Args:
+            yolo: A YOLOv8 model that's inherited from `:class:ultralytics.yolo.engine.model.YOLO`
+            run_name: The name of the run to be created on Weights & Biases.
+            project: The name of the project to be created on Weights & Biases.
+            tags: A list of tags to be added to the run on Weights & Biases.
+            resume: Whether to resume a previous run on Weights & Biases.
+            **kwargs: Additional arguments to be passed to `wandb.init()`.
+        """
+        self.yolo = yolo
+        self.run = None
+        self.run_name = run_name
+        self.project = project
+        self.tags = tags
+        self.resume = resume
+        self.kwargs = kwargs
+
+    def on_pretrain_routine_start(self, trainer: BaseTrainer):
+        """
+        Starts a new wandb run to track the training process and log to Weights & Biases.
+
+        Args:
+            trainer: A task trainer that's inherited from `:class:ultralytics.yolo.engine.trainer.BaseTrainer`
+                    that contains the model training and optimization routine.
+        """
+
+        if self.run is None:
+            if wandb.run is None:
+                self.run = wandb.init(
+                    name=self.run_name if self.run_name else trainer.args.name,
+                    project=self.project
+                    if self.project
+                    else trainer.args.project or "YOLOv8",
+                    tags=self.tags if self.tags else ["YOLOv8"],
+                    config=vars(trainer.args),
+                    resume=self.resume if self.resume else "allow",
```

Comment:
are you sure it is a good idea to default to resume? this code path has a lot of cases, and maybe we should let the user decide if they want to opt-in?

Reply:
Change this to default to `None` same as in `wandb.init`